torchani/utils.py

```diff
@@ -142,8 +143,31 @@ class EnergyShifter(torch.nn.Module):
 
     def __init__(self, self_energies):
         super(EnergyShifter, self).__init__()
-        self_energies = torch.tensor(self_energies, dtype=torch.double)
+
+        if self_energies is not None:
+            self_energies = torch.tensor(self_energies, dtype=torch.double)
+
         self.register_buffer('self_energies', self_energies)
+        self.present_species = present_species
+
+    def sae_from_dataset(self, atomic_properties, properties):
+        """Compute atomic self energies from dataset.
+
+        Least-squares solution to a linear equation is calculated to output
+        ``self_energies`` when ``self_energies = None`` is passed to
+        :class:`torchani.EnergyShifter`
+        """
+        species = atomic_properties['species']
+        energies = properties['energies']
+        present_species = self.present_species(species)
+        mask = []
```

Comment:
Instead of having a `mask` and doing a for loop, could we just do the following:
```python
X = (species.unsqueeze(-1) == present_species).sum(dim=1).to(torch.double)
```

Reply:
Fixed in dc85d3c